### ISSUE TITLE
Remove the autocomplete suggestions for the appengine repackaged classes

### ISSUE THREAD

**Issue by elharo** [GoogleCloudPlatform/google-cloud-eclipse#2534]
In Eclipse you can remove the autocomplete suggestions for certain packages (I do it manually for com.ibm.icu4j versions of stuff), you should do it for the appengine repackaged classes so developers don't get a bad autocomplete (since com.google.appengine is lexicographically before com.google.common.collect eclipse inadvertently autocompletes the wrong class, sometimes without prompting the user even

**Comment by briandealwis** [comment]
That should already be happening with the [`appengine-api`](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml#L22) library definition.

**Comment by chanseokoh** [comment]
The access rule does not seem to affect autocomplete suggestion.

**Comment by briandealwis** [comment]
Hmm, our code should be setting up these repackaged patterns [using an `IAccessRule.K_NONE_ACCESSIBLE` access rules](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/blob/master/plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/repository/LibraryClasspathContainerResolverService.java#L303).  The Java _Type Filters_ preference hides such forbidden references by default, though it allows `IAccessRule.K_DISCOURAGED` references (see pref page below).

So it sounds like we're either not configuring the access rules, or somehow our rules are allowing the repackaged classes to surface (I haven't verified this yet).  Or you perhaps you have the preference for forbidden elements disabled?

![screen shot 2017-10-31 at 11 29 28 am](https://user-images.githubusercontent.com/202851/32232644-e1940f54-be2e-11e7-96f9-93275375a134.PNG)

**Comment by briandealwis** [comment]
I don't see a way to easily add new type patterns to that list programmatically (`org.eclipse.jdt.internal.ui.preferences.TypeFilterPreferencePage#initialize()`.

**Comment by chanseokoh** [comment]
> So it sounds like we're either not configuring the access rules, or somehow our rules are allowing the repackaged classes to surface (I haven't verified this yet).

It is configured. (BTW, it's only for native projects.) They work in some sense, but in a sub-optimal way:
![selection_025](https://user-images.githubusercontent.com/10523105/32233587-18e37b14-be31-11e7-8bc2-e85a4314802e.png)

That is, the suggestion is basically empty in that sense that it does not suggest any class, but it doesn't prevent suggesting packages. However, because it does not suggest any class in the end (also such classes are strictly forbidden), so I think this is a minor issue.

However, we don't do anything for Maven projects. Currently, users can use all the classes that we want to forbid, so preventing them in auto-completion would be nice. (BTW, is there a way to exclude some classes from a dependency in Maven. I guess there may be.)

**Comment by briandealwis** [comment]
It is odd behaviour!  I've opened [a bug against JDT](https://bugs.eclipse.org/bugs/show_bug.cgi?id=526692).

**Comment by briandealwis** [comment]
I don't think we can really do anything for Maven users, beyond suggesting they add the `com.google.appengine.repackaged.*` to their JDT _Type Filters_ page.  The Maven Classpath Container is entirely managed by m2e, and it immediately overwrites manual attempts to add an access rule to its managed `appengine-sdk-1.0-XXX.jar`.
